.github/workflows/test.yml

```diff
@@ -6,10 +6,36 @@ on:
     types: [opened, synchronize]
     branches-ignore: [staging, production]
 
+env:
+  # Number of parallel jobs for jest tests
+  CHUNKS: 3
 jobs:
+  config:
+    runs-on: ubuntu-latest
+    name: Define matrix parameters
+    outputs:
+      MATRIX: ${{ steps.set-matrix.outputs.MATRIX }}
+    steps:
+      - name: Set Matrix
+        id: set-matrix
+        uses: actions/github-script@v6
+        with:
+          # Generate matrix array i.e. [0, 1, 2, ...., CHUNKS] for test job
+          script: |
+            core.setOutput('MATRIX', Array.from({ length: Number(process.env.CHUNKS) + 1 }, (v, i) => i));
```

Comment:
```suggestion
            core.setOutput('MATRIX', Array.from({ length: Number(process.env.CHUNKS) }, (v, i) => i));
```

So based on the latest test runs, four jobs are running. 
![image](https://user-images.githubusercontent.com/24370807/194394461-f0ef7933-3769-4a27-bf38-62c05efcad10.png)

So it would be good if only the number of jobs equal to the chunk value is triggered. I think you just need to remove `+ 1`.

Reply:
So if you want to create `n` jobs if `CHUNKS` value is `n` then jest based test will run in `n - 1` jobs since we need one job for  `Pull Request Tests`. Then that command will change from 
```
- name: Pull Request Tests
        # Pull request related tests will be run in separate runner in parallel.
        if: ${{ fromJSON(matrix.chunk) == fromJSON(env.CHUNKS) }}
        run: tests/unit/getPullRequestsMergedBetweenTest.sh
```

to 

```
- name: Pull Request Tests
        # Pull request related tests will be run in separate runner in parallel.
        if: ${{ fromJSON(matrix.chunk) == fromJSON(env.CHUNKS) - 1 }}
        run: tests/unit/getPullRequestsMergedBetweenTest.sh
```

And for jest based test, command will change from 
```
- name: All Unit Tests
        if: ${{ fromJSON(matrix.chunk) < fromJSON(env.CHUNKS) }}
         run: npx jest --listTests --json | jq -cM '[_nwise(length / ${{ fromJSON(env.CHUNKS) }} | ceil)]' | jq '.[${{ fromJSON(matrix.chunk) }}] | .[] | @text' | xargs npm test
```

to 

```
- name: All Unit Tests
        if: ${{ fromJSON(matrix.chunk) < fromJSON(env.CHUNKS) - 1 }}
        # Split the jest based test files in multiple chunks/groups and then execute them in parallel in different jobs/runners.
        run: npx jest --listTests --json | jq -cM '[_nwise(length / ${{ fromJSON(env.CHUNKS) }} | ceil)]' | jq '.[${{ fromJSON(matrix.chunk) }}] | .[] | @text' | xargs npm test
```


so for `n` equals `4`

matrix will be `[0, 1, 2, 3]` and jest based tests will run for `0`, `1`, `2` and for index `3`, pull request test will be run.

Let me know if that sound good then I can add this change.

Reply:
So I don't get this bug change. I was thinking if we just change the matrix creation logic, it will be fine. 

So 
```
chunks: 3
Current matrix logic : [0 , 1, 2, 3]
My suggestion: change it to  [0, 1, 2] (n-1) last element

then change the name to start from 1 instead of 0
```

Reply:
Never mind, It seems you are correct. I didn't notice that you said `so for n equals 4`. 

Please make the necessary changes.

Reply:
Is this change commited? I don't see this. Jobs should be equal to # of chunks. For our case, 3 jobs for 3 chunks.

Reply:
> Never mind, It seems you are correct. I didn't notice that you said `so for n equals 4`.
> 
> Please make the necessary changes.

I though you agreed with me. 

We in any case, we need to handle `n - 1` senario since we are using index for arrays of chunks. 

If you want 3 jobs for `CHUNKS: 3` then I've to add changes mentioned above like
```
jq -cM '[_nwise(length / ${{ fromJSON(env.CHUNKS) - 1 }}

if: ${{ fromJSON(matrix.chunk) == fromJSON(env.CHUNKS) - 1 }}

 if: ${{ fromJSON(matrix.chunk) < fromJSON(env.CHUNKS) - 1 }}
````

Reply:
yeah sure.

Reply:
Pushed changes

now CHUNKS: 3
Jest based tests will split into Job 1 and Job 2
Job 3 will execute PR tests